setup_script.sh

```diff
@@ -1,210 +1,126 @@
 #!/bin/bash
 
-###
-## ensure that required params are passed into script
-###
-if [ -n "$1" ]; then
-    HOST=$1
-else
-    echo "[ database ] : name not passed, set to default value (127.0.0.1)"
-    HOST="127.0.0.1"
-fi
-
-###
-## specify which database to operate on, if not specified, set default to camic
-###
-if [ -n "$2" ]; then
-    DB_NAME=$2
-    echo "[ database ] : name set as ${DB_NAME}"
-else
-    echo "[ database ] : name not passed, set to default value (camic)"
-    DB_NAME="camic"
-fi
-
-###
-## check if the system has required services installed
-###
-if ! command -v "mongo" &>/dev/null; then
-    echo "mongo could not be found on path. Please ensure that mongo is installed and is on PATH"
-    exit
-fi
-
-if ! command -v "node" &>/dev/null; then
-    echo "node could not be found on path. Please ensure that node is installed and is on PATH"
-    exit
-fi
-
-###
-# check for the existence of required files in the Distro and camicroscope repository
-###
-echo "Checking for required files"
-
-#if [[ ! -d ../Distro/db ]]
-#then echo "../Distro/db does not exist"
-#     exit
-#fi
-
-if [[ ! -d ../Distro/config ]]
-then echo "../Distro/config does not exist"
-     exit
-fi
-
-if [[ ! -d ../Distro/jwt_keys ]]
-then echo "../Distro/jwt_keys does not exist"
-     exit
-fi
-
-if [[ ! -d ../caMicroscope ]]
-then echo "../caMicroscope does not exist"
-     exit
-fi
+# Ensure required params are passed into the script
+HOST="${1:-127.0.0.1}"
+DB_NAME="${2:-camic}"
+
+# Function to check if a command exists
+check_command() {
+    if ! command -v "$1" &>/dev/null; then
+        echo "$1 could not be found on path. Please ensure that $1 is installed and is on PATH"
+        exit 1
+    fi
+}
+
+# Check for required commands
+check_command "mongoshsh"
+check_command "node.exe"
```

Comment:
Is this a windows only setup script?

Reply:
I have again updated the script based on linux the only difference is that for the windows machine we have to add extension of .exe to both command